src/datanode/src/config.rs

```diff
@@ -218,13 +216,8 @@ pub struct DatanodeOptions {
     pub node_id: Option<u64>,
     pub require_lease_before_startup: bool,
     pub init_regions_in_background: bool,
-    pub rpc_addr: String,
```

Comment:
It's an incompatible change. I'm not sure it's an acceptable refactor. If not, we can use some `#serde[]` annotations to make it compatible, for example, use `#[serde(flatten)]` and `#[serde(rename = "name")]`.

@evenyag @waynexia

Reply:
A breaking change may be better than `serde(flatten)`. But we must modify the PR title to mark it as breaking.

@shuiyisong @sunng87 What do you think?

Reply:
I agress. But instead of a hard breaking change, can we add a deprecation warning for some time to smooth the upgrade?

Reply:
If we want to deprecate it, one way is wrapping arguments like `rpc_addr` by `Option`. e.g. `rpc_addr: Option<String>`. We can use it to initialize the grpc option. We set them to None by default and print a warning if people set it.